### PR TITLE
[webpack plugins] Update plugins to support the new webpack 4 APIs.

### PR DESCRIPTION
**Pull request by iclanton**

### CODE REVIEW

File: webpack/resolve-chunk-plugin/src/ResolveChunkPlugin.ts [hunk 1]
```diff
@@ -1,76 +1,123 @@
...
+const PLUGIN_NAME: string = 'resolve-chunk-plugin';
 
 /**
  * See README for documentation on this plugin.
  *
  * @public
  */
 export class ResolveChunkPlugin implements Webpack.Plugin {
+  private _chunkIdMap: Map<string, ((id: number | undefined) => void)[]> =
+    new Map<string, ((id: number | undefined) => void)[]>();
```
Comment (pgonzal) [inline webpack/resolve-chunk-plugin/src/ResolveChunkPlugin.ts]: >(id: number | undefined) => void)[] [](start = 21, length = 35)

This expression is unwieldy.  `(id: number | undefined) => void)` should be given a name

**Comment by pgonzal** [comment]
> interface IRange { }

Maybe we should contribute some of these fixes back to DefinitelyTyped?

---
Refers to: webpack/resolve-chunk-plugin/typings/webpack-plugin-types.d.ts:4 in b3e0c35. [](commit_id = b3e0c35504bcb670987c27b314b4e2c180ae074f, deletion_comment = False)

**Comment by iclanton** [comment]
> interface IRange { }

These are some pretty esoteric types, but we can consider it. I'll leave a comment to revisit it when this code gets changed again.

---
In reply to: [395673833](https://github.com/Microsoft/web-build-tools/pull/683#issuecomment-395673833) [](ancestors = 395673833)

---
Refers to: webpack/resolve-chunk-plugin/typings/webpack-plugin-types.d.ts:4 in b3e0c35. [](commit_id = b3e0c35504bcb670987c27b314b4e2c180ae074f, deletion_comment = False)

File: webpack/resolve-chunk-plugin/src/ResolveChunkPlugin.ts [hunk 2]
```diff
@@ -1,76 +1,123 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+ * See LICENSE in the project root for license information.
+ */
+
 import * as Webpack from 'webpack';
```
[the remark this answers is not in the document]
Comment (iclanton) [reply webpack/resolve-chunk-plugin/src/ResolveChunkPlugin.ts]: Looks like things are good.

---
In reply to: [193973075](https://github.com/Microsoft/web-build-tools/pull/683#discussion_r193973075) [](ancestors = 193973075)

**Pull request merged**